engine/storage/snapshot/src/org/apache/cloudstack/storage/snapshot/StorageSystemSnapshotStrategy.java

```diff
@@ -92,7 +90,34 @@
 
     @Override
     public SnapshotInfo backupSnapshot(SnapshotInfo snapshotInfo) {
-        return snapshotInfo;
+
+        if (!snapshotInfo.getLocationType().equals(Snapshot.LocationType.SECONDARY)) {
```

Comment:
Since Enum values are guaranteed to be singletons, the `==` is that same as `equals`.  By changing this check to be ``snapshotInfo.getLocationType() != Snapshot.LocationType.SECONDARY``, a potential NPE is avoided and the code is more straightforward.